kolena/detection/_internal/model.py

```diff
@@ -97,6 +99,7 @@ def __init__(self, name: str, workflow: WorkflowType, metadata: Optional[Dict[st
 
     @classmethod
     @validate_arguments(config=ValidatorConfig)
+    @with_event(event_name=EventAPI.Event.CREATE_MODEL)
```

Comment:
We use these methods in `__init__` like 
```
try:
  load(..)
except NotFound:
  create(..)
```
Would this be adding confusion in analyzing the data?

Reply:
Do you mean that [this block](https://github.com/kolenaIO/kolena/blob/trunk/kolena/detection/_internal/model.py#L85-L90) will do the following:
1. If model name is present - only a `load_model` event is called
2. If model name is not preset - both `load_model` and `create_model` events are called

My opinion is that this is fine given that the client indeed tried to do both actions. Additionally, for (2), if the model is not present, then the event_api will [record the exception](https://github.com/kolenaIO/kolena/blob/1d93dcaf147723aa1a6aaead5f9cb06e39ad2c9b/kolena/_utils/instrumentation.py#L130) as `NotFoundError` for us to filter as needed

Reply:
Non-blocking comment: It'll be a bit of effort to analyze errors for `load` operations, as to if it's actually error or "expected".

Reply:
Agreed, I raised a similar point in an [earlier pr](https://github.com/kolenaIO/kolena/pull/293#issuecomment-1782995557). It comes down to if we want to have more distinct event names (`event-failed` and `event-succeeded`) vs fewer event names but with more metadata. We can modify things as we start to analyze if needed.